### PR TITLE
feat(CORSO-62) Delete a Gallery Photo

### PR DESCRIPTION
**Pull request by DenisaNicoletaH**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/browse/CORSO-62)
## Context:
Allows admins to delete images within the gallery

## Changes
- 90% Jacoco coverage
- Added delete button in EditGallery.tsx
- Added DeleteGalleryImage Selenide Test

## Before and After UI (Required for UI-impacting PRs)

Before: 
image

After:
image

### CODE REVIEW

File: corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/DeleteGalleryTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        driver.findElement(By.cssSelector(".row:nth-child(3) > .col-12:nth-child(1) .col-5")).click();
+        driver.findElement(By.cssSelector(".gallery-item:nth-child(2) > .mt-2")).click();
+        driver.findElement(By.cssSelector(".delete-galleryImage-btn > img")).click();
+        sleep(2000);
+        driver.findElement(By.cssSelector(".swal-button--confirm")).click();
+        sleep(2000);
+
+        driver.findElement(By.cssSelector(".swal-button")).click();
+        sleep(2000);
+        driver.quit();
```
Comment (KarinaSofia) [inline corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/DeleteGalleryTest.java:47]: Nit: You should consider adding more asserts, but looks good!

**Review by KarinaSofia — APPROVED** [review]
Great work!

File: corso_website_springboot/src/main/java/com/corso/springboot/Gallery_Subdomain/businesslayer/GalleryServiceImpl.java [hunk 2]
```diff
@@ -83,43 +86,80 @@ public GalleryResponse createGallery(GalleryRequest galleryRequest) throws IOExc
...
+        galleryRepository.findAll().stream()
+                .sorted(Comparator.comparingInt(Gallery::getImage_position))
+                .forEach(gallery -> {
+                    gallery.setImage_position(updatedGalleries.size() + 1);
+                    updatedGalleries.add(gallery);
+                });
+
+
+        return galleryRepository.saveAll(updatedGalleries);
+    }
```
Comment (Mila5847) [inline corso_website_springboot/src/main/java/com/corso/springboot/Gallery_Subdomain/businesslayer/GalleryServiceImpl.java:163]: Why did you add this method, why do we need it?

**Review by Mila5847 — APPROVED** [review]
Good job! Seems good and tested!

File: corso_website_springboot/src/main/java/com/corso/springboot/Gallery_Subdomain/businesslayer/GalleryServiceImpl.java (hunk 2 above)
Comment (DenisaNicoletaH) [reply corso_website_springboot/src/main/java/com/corso/springboot/Gallery_Subdomain/businesslayer/GalleryServiceImpl.java:163]: This is used to reorder the gallery image position because beforehand, when an image was deleted, the position wouldn't update so that it makes sense, hence this function.

File: corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/DeleteGalleryTest.java (hunk 1 above)
Comment (DenisaNicoletaH) [reply corso_website_springboot/src/test/java/com/corso/springboot/selenium_tests/DeleteGalleryTest.java:47]: Thank you! Yes, you are right. I did in fact forget to add asserts .I will keep it in mind in the future !
